tensorboard/components/tf_globals/globals-polymer.ts

```diff
@@ -14,10 +14,15 @@ limitations under the License.
 ==============================================================================*/
 import {customElement} from '@polymer/decorators';
 import {PolymerElement} from '@polymer/polymer';
+import {addToGlobals} from '../../webapp/angular_polymer_bridge';
 import * as tf_globals from './globals';
 
 @customElement('tf-globals')
 class TfGlobals extends PolymerElement {
   override _template = null;
   tf_globals = tf_globals;
+  constructor() {
+    super();
+    addToGlobals('setUseHash', tf_globals.setUseHash);
```

Comment:
I think it can be a lot simpler than this.

When one js binary (Our Polymer Binary) provides functionality that should be consumed by another js binary (Our Angular Binary), with property mangling turned on, it should (1) declare an API and (2) define the API and attach it to the window or object using unmangled symbols.

1) Declaring an API: I think this is largely taken care of by polymer_interop_types although if we wanted to make it more generally available we'd use TypeScript typings.

2) Defining the API and attaching it to the window unmangled: I think, for example, you could simply do this:

```
class TfGlobals extends PolymerElement {
  override _template = null;
  tf_globals = tf_globals;
  constructor() {
    super();
    // Attach the API without mangled names for cross-binary access.
    this['tf_globals'] = tf_globals;
    tf_globals['setUseHash'] = tf_globals.setUseHash;
  }
}
```

You can sort of prove this to yourself by renaming tf_globals' setUseHash to setUseHashInternal or 'blahBlahBlah' and it will still work for the angular binary to use TFGlobals.setUseHash.

Reply:
That is what I was doing initially and I think I definitely ended up overthinking it. I've returned to the simpler approach.